### PR TITLE
Introduce a new CoordinatorPlugin interface for native clusters

### PR DESCRIPTION
**Pull request by pdabre12**
## Description
Introduce a new CoordinatorPlugin interface for native clusters. 

## Test Plan
Unit tests

**Resolves :** https://github.com/prestodb/presto/issues/23297

### CODE REVIEW

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java [hunk 1]
```diff
@@ -852,6 +859,18 @@ private void installPlugin(Plugin plugin, boolean coordinatorOnly)
...
+    private void installCoordinatorPlugin(CoordinatorPlugin plugin, boolean coordinatorOnly)
+    {
+        long start = nanoTime();
+        for (TestingPrestoServer server : servers) {
+            if (coordinatorOnly && !server.isCoordinator()) {
+                continue;
+            }
+            server.installCoordinatorPlugin(plugin);
+        }
+        log.info("Installed plugin %s in %s", plugin.getClass().getSimpleName(), nanosSince(start).convertToMostSuccinctTimeUnit());
```
Comment (elharo) [inline presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java:871]: please delete this, passing tests should generate no output

Comment (tdcmeehan) [reply presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java:871]: Let's keep it consistent with the convention found elsewhere (just above)..  We can clean up this convention in a separate commit.

**Review by tdcmeehan — DISMISSED** [review]

File: presto-spi/src/main/java/com/facebook/presto/spi/CoordinatorPlugin.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import com.facebook.presto.spi.function.FunctionNamespaceManagerFactory;
+
+import static java.util.Collections.emptyList;
+
+public interface CoordinatorPlugin
```
Comment (tdcmeehan) [inline presto-spi/src/main/java/com/facebook/presto/spi/CoordinatorPlugin.java:20]: Can you add some Javadoc to describe this plugin?  ie. what it is used for, what differentriates it from Plugin?

File: presto-spi/src/main/java/com/facebook/presto/spi/CoordinatorPlugin.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import static java.util.Collections.emptyList;
+
+/**
+ * Introduces a new {@link CoordinatorPlugin} interface for native C++ clusters.
+ * Certain elements of the {@link Plugin} SPI are not used in Prestissimo deployments, or may result in inconsistencies.
+ * The {@link CoordinatorPlugin} includes only the interfaces relevant to native C++ clusters and
+ * is a successor to {@link Plugin} and will eventually replace it.
+ * It contains only interfaces that are valid and used in a coordinator.
+ */
+
```
Comment (tdcmeehan) [inline presto-spi/src/main/java/com/facebook/presto/spi/CoordinatorPlugin.java:27]:
```suggestion
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
One tiny nit, otherwise LGTM

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**